### DISCUSSION TITLE
How to use Config Context Data on Custom Links?

### DISCUSSION THREAD

**Discussion by joaolucasmacedo** [netbox-community/netbox#11937] [Q&A]
I use local context data for custom links. 
Let's say I have a few devices running HTTPS at a different port, like 10443. I set a local context data for each device and use `object.local_context_data.https_port` on my custom links to generate the appropriate URL. It is easy but I need to set the local config context for each device.
I can create a Config Context, so all devices of that device type would have the same config for https_port, but I can't use it on custom links.
My question is: If I the local config context name for custom links is `object.local_context_data`, what is the same for (global) config Contexts?

**Comment by candlerb** [comment]
* [Device inherits ConfigContextModel](https://github.com/netbox-community/netbox/blob/v3.4.5/netbox/dcim/models/devices.py#L440)
* In addition to [local_context_data](https://github.com/netbox-community/netbox/blob/v3.4.5/netbox/dcim/models/devices.py#L440) this provides accessor method [get_config_context](https://github.com/netbox-community/netbox/blob/v3.4.5/netbox/extras/models/configcontexts.py#L147-L168) which combines all the inherited config_context data with the local_context_data.

Therefore, what you need is  `{{ object.get_config_context().https_port }}` in your custom link.

**Comment by joaolucasmacedo** [comment]
Thanks @candlerb !
